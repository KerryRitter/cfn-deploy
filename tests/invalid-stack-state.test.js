const AWS = require('aws-sdk-mock');
const lib = require('..');


describe('CloudFormation.validateTemplate', () => {
  beforeAll(() => {
    AWS.mock('CloudFormation', 'validateTemplate', (params, callback) => {
      callback(null, true);
    });
    AWS.mock('CloudFormation', 'describeStacks', (params, callback) => {
      callback(null, {
        Stacks: [
          {
            StackStatus: 'UPDATE_IN_PROGRESS',
          },
        ],
      });
    });
    // AWS.mock('CloudFormation', 'createChangeSet', (params, callback) => {
    //   callback(new Error('INVALID_TEMPLATE'));
    // });
    // AWS.mock('CloudFormation', 'describeChangeSet', (params, callback) => {
    //   callback(new Error('INVALID_TEMPLATE'));
    // });
    // AWS.mock('CloudFormation', 'executeChangeSet', (params, callback) => {
    //   callback(new Error('INVALID_TEMPLATE'));
    // });
    // AWS.mock('CloudFormation', 'deleteChangeSet', (params, callback) => {
    //   callback(new Error('INVALID_TEMPLATE'));
    // });
    // AWS.mock('CloudFormation', 'waitFor', (params, callback) => {
    //   callback(new Error('INVALID_TEMPLATE'));
    // });
  });

  it('should error on invalid stack state', (done) => {
    lib({
      region: 'us-east-1',
      stackName: 'test-stack',
      template: './tests/templates/simple-s3-template.yaml',
    }).on('ERROR', (err) => {
      expect(err.message).toMatchSnapshot();
      done();
    });
  });

  afterAll(() => {
    AWS.restore('CloudFormation');
  });
});
