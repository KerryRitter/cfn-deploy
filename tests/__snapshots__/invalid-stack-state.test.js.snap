// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudFormation.validateTemplate should error on invalid stack state 1`] = `"Can't update stack when status is \\"UPDATE_IN_PROGRESS\\""`;
